### PR TITLE
Feature/automatic pythonpath

### PR DESCRIPTION
**Pull request by lasote**

### CODE REVIEW

**Comment by danimtb** [comment]
can this be merged @lasote ?

**Comment by lasote** [comment]
If you like it... LoL

File: reference/tools.rst [hunk 1]
```diff
@@ -478,13 +478,18 @@ Parameters:
...
     def pythonpath(conanfile)
 
 This is a context manager that allows to load the PYTHONPATH for dependent packages, create packages
 with python code, and reuse that code into your own recipes.
 
+It is automatically applied
+
 .. code-block:: python
 
     from conans import tools
```
Comment (danimtb) [inline reference/tools.rst]: If I understood this feature well, using `tools.pythonpath()` is no longer needed if `apply_env `is `True`. So this code

```
def build(self):
    with tools.pythonpath(self):
        from module_name import whatever
        whatever.do_something()
```
can still be used even if `apply_env` is `True` and would be the same as typing:

```
def build(self):
    from module_name import whatever
    whatever.do_something()
```
is that right?

Comment (lasote) [reply reference/tools.rst]: True. Do you want me to add these comments?

**Comment by danimtb** [comment]
It would be nice to have them, yes! :D

2018-02-13 9:46 GMT+01:00 Luis Martinez de Bartolome Izquierdo <
notifications@github.com>:

**Comment by lasote** [comment]
Done

**Pull request merged**